lib/url.js

```diff
@@ -192,6 +194,18 @@ Url.prototype.parse = function(url, parseQueryString, slashesDenoteHost) {
     if (atSign !== -1) {
       auth = rest.slice(0, atSign);
       rest = rest.slice(atSign + 1);
+
+      var username, password;
+      colon = auth.lastIndexOf(':');
+      if (colon !== -1) {
+        username = auth.slice(0, colon);
```

Comment:
What if the username has no colons but the password contains one or more colons? What if both of them have colons? FWIW Chrome 45 and Firefox 40 split on the first colon instead of the last. I think it might be better to follow what browsers are doing so that at least there is some similarity.